authentication/src/main/java/io/camunda/authentication/config/WebSecurityConfig.java

```diff
@@ -129,50 +128,117 @@ private void genericSuccessHandler(
-                              .includeSubDomains(true)
-                              .maxAgeInSeconds(63072000)
-                              .preload(true)))
-          .exceptionHandling(
-              (exceptionHandling) -> exceptionHandling.accessDeniedHandler(authFailureHandler))
-          .csrf(AbstractHttpConfigurer::disable)
-          .cors(AbstractHttpConfigurer::disable)
-          .formLogin(AbstractHttpConfigurer::disable)
-          .anonymous(AbstractHttpConfigurer::disable);
-    } catch (final Exception e) {
-      throw new RuntimeException(e);
-    }
+  @Bean
+  @Profile("auth-basic")
+  @Order(2)
+  public SecurityFilterChain loginAuthSecurityFilterChain(
+      final HttpSecurity httpSecurity, final AuthFailureHandler authFailureHandler)
+      throws Exception {
+    return buildLoginAuthSecurityFilterChain(httpSecurity, authFailureHandler);
+  }
+
+  public SecurityFilterChain buildLoginAuthSecurityFilterChain(
+      final HttpSecurity httpSecurity, final AuthFailureHandler authFailureHandler)
+      throws Exception {
+    LOG.info("Configuring login auth");
+    return baseHttpSecurity(httpSecurity, authFailureHandler)
+        .formLogin(
+            formLogin ->
+                formLogin
+                    .loginPage(LOGIN_URL)
+                    .loginProcessingUrl(LOGIN_URL)
+                    .failureHandler(authFailureHandler)
+                    .successHandler(this::genericSuccessHandler))
+        .logout(
+            (logout) ->
+                logout
+                    .logoutUrl(LOGOUT_URL)
+                    .logoutSuccessHandler(this::genericSuccessHandler)
+                    .deleteCookies(SESSION_COOKIE))
+        .exceptionHandling(
+            exceptionHandling ->
+                exceptionHandling
+                    .authenticationEntryPoint(authFailureHandler)
+                    .accessDeniedHandler(authFailureHandler))
+        .build();
   }
 
   @Bean
   public FilterRegistrationBean<TenantRequestAttributeFilter>
       tenantRequestAttributeFilterRegistration(final MultiTenancyConfiguration configuration) {
     return new FilterRegistrationBean<>(new TenantRequestAttributeFilter(configuration));
   }
+
+  private static boolean isBasicAuthRequest(final HttpServletRequest request) {
+    final String authorizationHeader = request.getHeader(HttpHeaders.AUTHORIZATION);
+    // If the client sends an `Authorization: Basic` header, treat this as a basic auth request.
+    if (authorizationHeader != null && authorizationHeader.startsWith("Basic ")) {
+      return true;
+    }
+    // If it's an API request and there's no session cookie, also treat the request as a basic auth
+    // request. This is a workaround for Java's HTTP client, which only sends an Authorization
+    // header after the server sends a WWW-Authenticate header in a 401 response.
+    if (isApiRequest(request) && !hasSessionCookie(request)) {
+      return true;
+    }
+    return false;
```

Comment:
```suggestion
return isApiRequest(request) && !hasSessionCookie(request);
```

Reply:
I'd prefer to keep the structure the same for the two conditions, but I don't feel strongly about it. This whole method is pulled apart in the next PR anyway: https://github.com/camunda/camunda/pull/26925

Reply:
Ah if we're working with the method in the next PR then lets just ignore this point for now :)